Update simpleice, bring it closer to simpleice2

fjansson opened the pull request.

fjansson committed:
modsimpleice: fix uninitialized sgratio, loop up to kmax

fixes to bring modsimpleice closer to modsimpleice2.
Here fix uses of uninitialized variables.

* sgratio,rsgratio were calculated only for cells with qr.
In the sedimentation time splitting loop, those ratios were used also
for cells that initially did not have qr. The ratios are now calculated for
all cells.

* changed the process loops to go up to kmax, not k1.
which for modbulkmicro improved the stability with open boundary conditions.

fjansson committed:
modsimpleice: change snow ventilation factors in evapdep

!Grabovski 1998 https://doi.org/10.1175/1520-0469(1998)055<3283:TCRMOL>2.0.CO;2 says
F  = 0.78 + 0.27 Re^1/2  for raindrops
F  = 0.65 + 0.39 Re^1/2  for ice particles

this brings modsimpleice in line with modsimpleice2.

fjansson committed:
modsimpleice: add timer calls

fjansson committed:
modsimpleice2: add timer calls

fjansson committed:
modsimpleice: add timer calls for setup and finalization part of main routine

fjansson committed:
modsimpleice: switch to using modbulkmicrotend

The init of modsimpleicetend was never called, so it didn't work.
Modbulkmicrotend seems to work as a replacement, also for tendencies.
This commit removes simpleicetend and makes simpleice use
bulkmicrotend. Possible downside is that bulkmicrotend calculates
statistics also of Nr which simpleice doesn't use.

CasparJungbacker commented on src/modsimpleice.f90:
```diff
@@ -131,6 +132,8 @@ subroutine simpleice
     real:: qrsmall, qrsum,qrtest
 
+    call timer_tic('modsimpleice', 1)
+    call timer_tic('modsimpleice/setup', 1)
```
Just a suggestion: I've recently started giving every module and subroutine a character parameter for their name. This makes calling the timers easier and less error-prone. E.g:

```fortran
module modsimpleice
  character(len=*), parameter :: modname = "modsimpleice"
contains
  subroutine simpleice
    character(len=*), parameter :: routine = modname//"/simpleice"
    call timer_tic(routine, 1)
    ...
    call timer_toc(routine) <--- automatically the correct argument
  subroutine simpleice
end module modsimpleice
```

fjansson committed:
modsimpleice: clip negative qr like in modbulkmicro, sample clipping tendency

fjansson committed:
modsimpleice: make fudge factors parameters and decrease them.

both fudge factors (used to avoid /0) now set to qrmin = 1e-13.

fjansson committed:
modsimpleice and modbulkmicostat: fix missing initializations

fjansson committed:
modsimpleice: add some field_r, variable precision fixes

fjansson committed:
modstartup: initialize u0av,v0av,thl0av,qt0av,sv0av before first call to boundary

u0av, v0av are used in large scale forcings
thl0av, qt0av, sv0av are used to set values at k1 in modboundary

fjansson committed:
modbulkmicrostat: add qtsed, qt-tendency from cloud sedimentation

fjansson committed:
modsimpleice: tabulate an expression with sqrt for evapdep

fjansson committed:
modsimpleice: remove qrmask, qcmask fields

fjansson committed:
modsimpleice: move more variables and constants to configurable precision

fjansson committed:
modsimpleice: timer tic_toc with module and routine names as parameters
